File: twitterClient/twitterClient/src/stomp/frames/SubscribeFrame.cpp
//
//  SubscribeFrame.cpp
//  Object to represent the SUBSCRIBE frame type
//
//  Authors:    Ken Saggy
//              Lidan Hifi
//
//  Copyright 2013 Lidan Hifi, Ken Saggy

#include "../../../include/stomp/frames/SubscribeFrame.h"

namespace stomp {
    SubscribeFrame::SubscribeFrame(std::string destination, std::string identifier): StompFrame(SUBSCRIBE_FRAME_NAME) {
        addHeaderPair("destination", destination);
        addHeaderPair("id", identifier);
    }
    
    void SubscribeFrame::accept(FrameVisitor *visitor)
    { }
}

File: twitterEngine/src/spl/stomp/protocol/SubscribeFrame.java
package spl.stomp.protocol;

/**
 * @name Subscribe Frame
 * @author  Lidan Hifi
 * @author  Ken Saggy
 * 
 * SUBSCRIBE stomp frame
 */
class SubscribeFrame extends StompFrame {
    public SubscribeFrame() {
        super(StompCommand.SUBSCRIBE);
    }
    
    public SubscribeFrame(String destination, String id) {
        super(StompCommand.SUBSCRIBE);
        addFrameHeader("destination", destination);
        addFrameHeader("id", id);
    }
    
    @Override
    protected String[] getMandatoryHeaders() {
        return new String[] {"destination", "id"};
    }
}
